### DISCUSSION TITLE
Calculate particles "inside" in relation to submarine

### DISCUSSION THREAD

**Discussion by UnhappyWithMyUsername** [FakeFishGames/Barotrauma#15211] [Bug reports]
Particles always seem to be calculated in relation to world position, even if the particle is inside the submarine, thus technically in a completely different space. This causes particles to shift and jitter all over the place when the submarine moves.
Wouldn't it make more sense to calculate their position in relation to the submarine rather than world position to have these particles behave as expected and not look weird?

The same also goes for sounds. Non-looped sound effects currently drift off-screen when the submarine moves.

**Comment by girlyguppy** [comment]
Real

**Comment by Ek-Videogames** [comment]
Yeah, it's particularly noticeable with flares, pumps, and fire effects.

**Comment by Regalis11** [comment]
I don't think it should be necessary to calculate the particles in the "inside" space. That would incur some performance cost, because the draw positions of the particles would need to be calculated on the fly based on the position of the sub.

But there definitely seems to be something off with the particles that's causing this sort of jitter, I believe it could have something to do with the interpolated draw positions not being used correctly at some point in the particle rendering or movement logic. Moving them to the "internal space" should not be required to fix that though.

**Comment by Regalis11** [comment]
Converting this into a bug ticket, seems like no-one's reported this issue before. :D

**Comment by Rokvach** [comment]
Fixed and tested the fix to be working correctly, fix coming in the next update. Closing discussion.
